mtp_api/apps/transaction/tests/test_bank_admin_views.py

```diff
@@ -551,3 +542,56 @@ def test_get_list_excluding_invalid_label_includes_all(self):
         result_ids = [t['id'] for t in results]
         for trans in refunded_trans:
             self.assertTrue(trans.id in result_ids)
+
+
+class GetTransactionsFilteredByDateTestCase(
+    TransactionRejectsRequestsWithoutPermissionTestMixin,
+    BaseTransactionViewTestCase
+):
+    ENDPOINT_VERB = 'get'
+
+    def setUp(self):
+        super().setUp()
+
+        # delete all transactions and logs
+        Transaction.objects.all().delete()
+        Log.objects.all().delete()
+
+        self._populate_transactions()
+
+    def _get_unauthorised_application_users(self):
+        return [
+            self.prison_clerks[0], self.prisoner_location_admins[0]
+        ]
+
+    def _get_url(self, *args, **kwargs):
+        return reverse('bank_admin:transaction-list')
+
+    def _populate_transactions(self, tot=80):
+        transactions = generate_transactions(transaction_batch=tot)
+
+    def _get_authorised_user(self):
+        return self.bank_admins[0]
+
+    def test_get_list_received_between_dates(self):
+        url = self._get_url()
+        user = self._get_authorised_user()
+
+        response = self.client.get(
+            url, {'received_at__lt': date.today(),
+                  'received_at__gte': date.today() - timedelta(days=2),
+                  'limit': 1000}, format='json',
+            HTTP_AUTHORIZATION=self.get_http_authorization_for_user(user)
+        )
+        self.assertEqual(response.status_code, http_status.HTTP_200_OK)
+
+        results = response.data['results']
+        result_ids = [t['id'] for t in results]
+        received_between_dates = Transaction.objects.filter(
```

Comment:
fyi: this filter raises naive date-time warning. not a big deal tbh; it's only a test :)